### PR TITLE
Add topics

### PR DESCRIPTION
**Pull request by alealvarez14**
This PR includes:

- MVC for Topics
- Admin page to create topics with images
- Test for Topics

### CODE REVIEW

File: app/decorators/topic_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,3 @@
+class TopicDecorator < Draper::Decorator
```
Comment (Alxzu) [inline app/decorators/topic_decorator.rb]: Are you using this?

File: spec/requests/api/v1/topics/index_spec.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
+require 'rails_helper'
+
+describe 'GET api/v1/topics/', type: :request do
+  let(:user) { create(:user) }
+  let(:api_v1_topic_path) { '/api/v1/topics' }
+
+  context "only one topic exists" do
```
Comment (Alxzu) [inline spec/requests/api/v1/topics/index_spec.rb]: Single quotes

File: app/controllers/api/v1/topics_controller.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+module Api
+  module V1
+    class TopicsController < Api::V1::ApiController
+      def index
+        @topics = Topic.all.includes(:icon_attachment)
```
Comment (GuilleLeopold) [inline app/controllers/api/v1/topics_controller.rb]: you can just do `@topics = Topic.includes(:icon_attachment)`

File: app/decorators/topic_decorator.rb (hunk 1 above)
Comment (GuilleLeopold) [inline app/decorators/topic_decorator.rb]: You are not using this, right?

File: spec/requests/api/v1/topics/index_spec.rb [hunk 4]
```diff
@@ -0,0 +1,33 @@
+require 'rails_helper'
+
+describe 'GET api/v1/topics/', type: :request do
+  let(:user) { create(:user) }
+  let(:api_v1_topic_path) { '/api/v1/topics' }
```
Comment (Alxzu) [inline spec/requests/api/v1/topics/index_spec.rb:5]: This is great, you should take a look at [subjects](https://stackoverflow.com/questions/38437162/whats-the-difference-between-rspecs-subject-and-let-when-should-they-be-used) and get your own conclusions if it works for you

**Review by Alxzu — COMMENTED** [review]
There are a couple of places where you are using double quotes, but really good first PR 💪 

[betterspecs](http://www.betterspecs.org/)

File: app/models/topic.rb [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+#  title      :string
+#  icon       :string
+#  created_at :datetime         not null
+#  updated_at :datetime         not null
+#
+
+class Topic < ApplicationRecord
+  validates :title, presence: true
+
+  has_one_attached :icon
```
Comment (GuilleLeopold) [inline app/models/topic.rb:13]: Associations and then validations. It's just a matter of keeping the same structure

File: app/views/api/v1/topics/index.json.jbuilder [hunk 6]
```diff
@@ -0,0 +1,3 @@
+json.topics @topics.each do |topic|
+  json.partial! 'api/v1/topics/info', topic: topic
```
Comment (GuilleLeopold) [inline app/views/api/v1/topics/index.json.jbuilder]: just 'info' it's ok as you are in the same folder

File: spec/models/topic_spec.rb [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+#  icon       :string
+#  created_at :datetime         not null
+#  updated_at :datetime         not null
+#
+
+require 'rails_helper'
+
+describe Topic do
+  describe 'validations' do
+    it { should validate_presence_of(:title) }
```
Comment (GuilleLeopold) [inline spec/models/topic_spec.rb]: instead of using should, use is_expected

File: spec/support/file_test_helper.rb [hunk 8]
```diff
@@ -0,0 +1,14 @@
+module FilesTestHelper
+  extend self
+  extend ActionDispatch::TestProcess
+
+  def png_name; 'test-image.png' end
```
Comment (GuilleLeopold) [inline spec/support/file_test_helper.rb]: Use only this kind of syntax when the method is empty

File: app/decorators/topic_decorator.rb (hunk 1 above)
Comment (alealvarez14) [reply app/decorators/topic_decorator.rb]: No, I'll delete it

Comment (alealvarez14) [reply app/decorators/topic_decorator.rb]: No, I'll delete it

File: spec/support/file_test_helper.rb (hunk 8 above)
Comment (alealvarez14) [reply spec/support/file_test_helper.rb]: Fixed

**Review by Alxzu — APPROVED** [review]
👏

File: db/migrate/20190408182315_create_topics.rb [hunk 9]
```diff
@@ -0,0 +1,10 @@
+class CreateTopics < ActiveRecord::Migration[5.2]
+  def change
+    create_table :topics do |t|
+      t.string :title
```
Comment (GuilleLeopold) [inline db/migrate/20190408182315_create_topics.rb]: missing `null: false`

File: spec/requests/api/v1/topics/index_spec.rb (hunk 4 above)
Comment (GuilleLeopold) [reply spec/requests/api/v1/topics/index_spec.rb:5]: subject works fine here as the request is always the same

**Review by GuilleLeopold — APPROVED** [review]

File: db/migrate/20190408182315_create_topics.rb (hunk 9 above)
Comment (alealvarez14) [reply db/migrate/20190408182315_create_topics.rb]: Fixed

**Pull request merged**